test/test_jit_forms.py

```diff
@@ -1404,3 +1405,149 @@ def test_ds_prism(compile_args, dtype):
+@pytest.mark.parametrize("rank", [0, 1, 2])
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        np.float32,
+        np.float64,
+        pytest.param(
+            np.complex64,
+            marks=pytest.mark.skipif(
+                os.name == "nt", reason="win32 platform does not support C99 _Complex numbers"
+            ),
+        ),
+        pytest.param(
+            np.complex128,
+            marks=pytest.mark.skipif(
+                os.name == "nt", reason="win32 platform does not support C99 _Complex numbers"
+            ),
+        ),
+    ],
+)
+def test_vertex_integral(compile_args, geometry, rank, dtype):
+    cell, gdim = geometry
+    rdtype = np.real(dtype(0)).dtype
+    domain = ufl.Mesh(basix.ufl.element("Lagrange", cell, 1, shape=(gdim,), dtype=rdtype))
+    element = basix.ufl.element("Lagrange", cell, 1)
+    dP = ufl.Measure("dP")
+    x = ufl.SpatialCoordinate(domain)
+    V = ufl.FunctionSpace(domain, element)
+    u, v = ufl.TrialFunction(V), ufl.TestFunction(V)
+    # With a Lagrange space for test and trial functions, all these forms should evaluate as the
+    # x-coordinate times an indicator for alignment between the argument(s) and the vertices.
+    if rank == 0:
+        F = x[0] * dP
+    elif rank == 1:
+        F = x[0] * ufl.conj(v) * dP
+    else:
+        F = x[0] * u * ufl.conj(v) * dP
+
+    (form0,), module, _ = ffcx.codegeneration.jit.compile_forms(
+        [F], options={"scalar_type": dtype}, cffi_extra_compile_args=compile_args
+    )
+    assert form0.rank == rank
+
+    ffi = module.ffi
+    kernel = getattr(
+        form0.form_integrals[0],
+        f"tabulate_tensor_{dtype().dtype.name}",
+    )
+
+    vertex_count = domain.ufl_coordinate_element().basix_element.points.shape[0]
+
+    for a, b in np.array([(0, 1), (1, 0), (2, 0), (5, -2)], dtype=dtype):
+        # General geometry for simplices of gdim 1,2 and 3.
+        # gdim 1: creates the interval (a, b)
+        # gdim 2: creates the triangle with vertices (a, 0), (b, 0), (0,0)
+        # gdim 3: creates the tetrahedron with vertices (a, 0, 0), (b, 0, 0), (0, 0, 0), (0, 0, 1)
+        coords = np.array([a, 0.0, 0.0, b, 0.0, 0.0, 0, 0, 0, 0, 0, 1], dtype=rdtype)
+
+        for vertex in range(vertex_count):
+            J = np.zeros(vertex_count**2, dtype=dtype)
+            e = np.array([vertex], dtype=np.int32)
+            kernel(
+                ffi.cast(f"{dtype_to_c_type(dtype)} *", J.ctypes.data),
+                ffi.NULL,
+                ffi.NULL,
+                ffi.cast(f"{dtype_to_c_type(rdtype)} *", coords.ctypes.data),
+                ffi.cast("int *", e.ctypes.data),
+                ffi.NULL,
+                ffi.NULL,
+            )
+            idx = (rank > 0) * vertex + (rank > 1) * vertex * vertex_count
+            assert np.isclose(J[idx], coords[vertex * 3], atol=1e2 * np.finfo(dtype).eps)
+            assert np.allclose(np.delete(J, [idx]), 0)
```

Comment:
```suggestion
            # Check all other entries are not touched
            assert np.allclose(np.delete(J, [idx]), 0)
```
I think I'd add such comment for clarity.